### DISCUSSION TITLE
When adding an element to a list-structured db, does TinyDB write the entire json to disc or just append the new item?

### DISCUSSION THREAD

**Discussion by bipsen** [msiemens/tinydb#408] [Q&A]
Hi,

I am considering TinyDB for a usecase where I am receiving a json-file from an API every second or so and have to contiously append to a json file. I have previously used a simple list-formatted json file for this, but when the files get big I face the problem of having to load the entire json file into memory, append the new element, and then save it again.

I am wondering if TinyDB actually saves me from having to load the entire json into memory and write all the data on each iteration. 

Thank you for an awesome library!

**Comment by msiemens** [comment]
Hey @bipsen, I fear that in your situation TinyDB isn't really able to help you. It's designed for small datasets that can quickly be loaded from disk. As a result of this design choice, TinyDB does indeed load the data from disk on every read operation. There's a workaround using the [`CachingMiddleware`](https://tinydb.readthedocs.io/en/latest/usage.html#cachingmiddleware) that keeps the data in memory and only reads/writes data to disk after a specified number of operations. But if you have trouble with loading your whole dataset into memory that won't be useful either.

But if you can use something like the [JSON Lines](https://jsonlines.org/) format, you could simply open the file with append mode and write the JSON result to a new line to the end of your file. That would solve the issue of having to load the whole file for every read operation. But I don't know if this would work for your particular circumstances.

**Reply by msiemens** [reply]
@bipsen By the way, I've moved this into the discussions section of the project. Just so you know why the original issue is marked as closed 🙂
